Core/include/Acts/Geometry/DetectorElementBase.hpp

```diff
@@ -31,23 +32,35 @@ class Surface;
 
 class DetectorElementBase {
  public:
-  DetectorElementBase() = default;
-  virtual ~DetectorElementBase() = default;
+  /// Construct a detector element. It will have ownership of the surface
+  DetectorElementBase(std::shared_ptr<Surface>&& surface) : m_surface(surface) {
+    if (m_surface->m_associatedDetElement != nullptr) {
+      throw std::runtime_error("Detector Element already associated");
+    }
+
+    m_surface->m_associatedDetElement = this;
+  }
+  virtual ~DetectorElementBase() {
+    m_surface->m_associatedDetElement = nullptr;
+  }
```

Comment:
I wonder if this is strictly necessary. Also could that be a problem with thread safety?

Reply:
Yes, and no, actually `DetectorElement` should live longer than even the `MultiTrajectory`, but it is hard to control.